research/zkLend_liquidation_research.md

```diff
@@ -0,0 +1,52 @@
+# Research on zkLend Liquidation
+
+This research explored how to check and verify that zkLend closed a position due to liquidation rather than by a user through the Spotnet smart contract, particularly focusing on the deployed spotnet smart contract, the events emitted by the zkLend contract and the state changes associated with liquidation.
+
+## Introduction
+
+zkLend liquidation process is triggered when a borrower's Health Factor falls to 1 or lower, indicating that
+the value of their collateral may not be enough to cover their debt, posing risks to the lenders and zkLend
+protocol. The liquidation process incentivises external party(ies) to actively monitor zkLend's loan
+portfolios to seek out borrowings that have Health Factor below 1 and to actively repay these loans to
+ensure liquidity safety and health of the protocol. 
+
+A borrower whose outstanding loan exceeds his/her borrowing capacity will be liquidated at market rate.
+Instead of a static close factor, zkLend will implement a variable close factor where liquidators can only
+repay the portion (or part) of loan position that has exceeded the borrowing capacity. While the liquidation
+threshold for each borrower varies on his/her assets pledged as well as the borrowing asset, liquidations
+happen when the underlying collateral decreases in value relative to the borrowed asset and/or if the
+outstanding loan value (asset value + accrued interest) now exceeds the borrowing capacity. Examples
+may include large price increases in a borrowed asset, or dramatic price decrease in the underlying
+collateral.
+
+Upon reaching liquidation threshold, any Starknet address (i.e. Liquidator) can initiate a liquidationCall()
+contract to repay the debt on behalf of the borrower and receive a Liquidation Bonus (of usually 10%). 
+
+## Research Methodology
+
+To verify if a position was liquidated in zkLend, the following steps were taken:
+
+### User Activity
+1. **Check Events in Deployed Spotnet Contract**: The `Events` logs of the spotnet contract were examined on [Starkscan](https://starkscan.co/contract/0x05685d6b0b493c7c939d65c175305b893870cacad780842c79a611ad9122815f#events). Only two events were observed; `LiquidityLooped` and `OwnershipTransferred`
+2. **[Check Read Functions](https://starkscan.co/contract/0x05685d6b0b493c7c939d65c175305b893870cacad780842c79a611ad9122815f#read-write-contract-sub-read)**: `is_position_open()` returned 1 (true), while `owner()` returned 0x5c0846b4a80bb664b2f865e4dbc9a5e5eb3c454d124124ab891acc55a7e6fd.
```

Comment:
I don't think `owner` is relevant for this research

Reply:
Yes, initially it'll be useful, it is not. Will remove it.